### PR TITLE
Steel 89 subnet resource and datasource

### PR DESCRIPTION
**Pull request by mastamark**

### CODE REVIEW

File: website/docs/r/cm_subnet.markdown [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+layout: "rightscale"
+page_title: "Rightscale: subnet"
+sidebar_current: "docs-rightscale-resource-subnet"
+description: |-
+  Create and maintain a rightscale cloud management subnet.
+---
+
+# rightscale_subnet
+
+Use this resource to create, update or destroy rightscale [subnets](http://reference.rightscale.com/api1.5/resources/ResourceSubnets.html).
```
Comment (raphael) [inline website/docs/r/cm_subnet.markdown]: RightScale

File: website/docs/r/cm_subnet.markdown [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+output "us-oregon-devops-subnet-a-aws-href" {
+  value = "${rightscale_network.devops-oregon-subnet-a.href}"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `cloud_href` - (Required) Cloud you want to create the subnet in.
```
Comment (raphael) [inline website/docs/r/cm_subnet.markdown]: Href of

File: website/docs/r/cm_subnet.markdown [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+* `network_href` - (Required) ID of network to create subnet in.
```
Comment (raphael) [inline website/docs/r/cm_subnet.markdown]: Href of

File: website/docs/r/cm_subnet.markdown [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+* `cidr_block` - (Required) Subnet allocation range in CIDR notation.
+
+* `name` - (Optional) Subnet name.
+
+* `description` - (Optional) Subnet description.
+
+* `datacenter_href` - (Optional) Cloud datacenter to assign subnet to.
```
Comment (raphael) [inline website/docs/r/cm_subnet.markdown]: Href of

File: website/docs/r/cm_subnet.markdown (hunk 1 above)
Comment (mastamark) [reply website/docs/r/cm_subnet.markdown]: Will fix up in big cleanup before 1.0 release.

File: website/docs/r/cm_subnet.markdown [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+The following attributes are exported:
+
+* `href` - Href of the subnet.
+
+* `resource_uid` - Cloud resource_uid.
+
+* `is_default` - Indicates whether the subnet is the network default subnet. (true or false)
+
+* `state` - Indicates whether subnet is pending, available etc.
```
[the remark this answers is not in the document]
Comment (mastamark) [reply website/docs/r/cm_subnet.markdown]: Me too - any idea how to figure out what those options are as the api docs are lacking this info (http://reference.rightscale.com/api1.5/media_types/MediaTypeSubnet.html) - or is this perhaps a per-cloud thing so we really have no idea what cloud would report what states here?

**Pull request merged**